src/__readlink_chk.c

```diff
@@ -41,7 +42,12 @@ wrapper(__readlink_chk, ssize_t, (const char * path, char * buf, size_t bufsiz,
     debug("__readlink_chk(\"%s\", &buf, %zd, %zd)", path, bufsiz, buflen);
     expand_chroot_path(path);
 
-    if ((linksize = nextcall(__readlink_chk)(path, tmp, FAKECHROOT_PATH_MAX-1, buflen)) == -1) {
+    if (__builtin_expect(!!(bufsiz > buflen), 0)) {
+      printf("readlink: prevented write past end of buffer\n");
+      exit(-1);
```

Comment:
why the exit? why not just return failure?

Reply:
Because `_chk` functions should `exit` on overflow. Actually `__chk_fail` should be called (similar to `__realpath_chk`).
https://refspecs.linuxbase.org/LSB_4.1.0/LSB-Core-generic/LSB-Core-generic/libc---readlink-chk-1.html